### PR TITLE
refactor(service): improve strategy pattern for PredictionService class

### LINKED ISSUE #42: [service] Improve the strategy pattern for selecting the Italian or English NLU model

### PR DESCRIPTION
**Pull request by kevinpcf**
* create a new strategy for model selection
* extract PredictionService class from CADOCS
* modify LanguageModel class to an abstract class

Closes #42

### CODE REVIEW

File: cadocs_service.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import re
 import flask
 import CADOCS
```
Comment (alfcan) [inline cadocs_service.py]: this import is not used

File: prediction_service.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from model_selector import ModelSelector
+from CADOCS import ItalianModel, EnglishModel
+from langdetect import detect
+
+# This is the class that supports language detection
+class PredictionService:
+    def predict(self, message):
+        if self._is_italian(message):
+            model = ModelSelector(ItalianModel())
```
Comment (davide-lagamba) [inline prediction_service.py:9]: The ModelSelector could be instantiated outside this function and here use the setter methods

**Review by davide-lagamba — CHANGES_REQUESTED** [review]
PredictionService class could be modified instantiating ModelSelector outside the predict method and using the setter methods

**Review by alfcan — CHANGES_REQUESTED** [review]
CADOCS import into cadocs_service is not used

File: prediction_service.py (hunk 2 above)
Comment (kevinpcf) [reply prediction_service.py:9]: I chose to instantiate the model in the if statement in order to improve code readability

**Review by alfcan — APPROVED** [review]

File: cadocs_service.py (hunk 1 above)
Comment (kevinpcf) [reply cadocs_service.py]: I done a commit for resolve this

**Review by davide-lagamba — APPROVED** [review]

**Pull request merged**